src/shogun/preprocessor/RandomFourierGaussPreproc.h

```diff
@@ -91,147 +61,68 @@ class RandomFourierGaussPreproc: public RandomMixin<DensePreprocessor<float64_t>
-	 *
+	/** setter for output feature space dimension
+	 * @param dim feature space dimension
 	 */
 	void set_dim_feature_space(const int32_t dim);
 
-	/** computes new random coefficients IF test_rfinited() evaluates to false
-	 * test_rfinited() evaluates to TRUE if void set_randomcoefficients(...)
-	 * hase been called and the values set by set_dim_input_space(...) ,
-	 * set_dim_feature_space(...) and set_kernelwidth(...) are consistent to the
-	 * call of void set_randomcoefficients(...)
-	 *
-	 * throws shogun exception if dim_feature_space <= 0 or dim_input_space <= 0
-	 *
-	 * @return returns true if test_rfinited() evaluates to false and new
-	 * coefficients are computed
-	 * returns false if test_rfinited() evaluates to true and old random
-	 * coefficients are kept which were set by a previous call to void
-	 * set_randomcoefficients(...)
-	 *
-	 * this function is useful if you want to use apply_to_feature_vector but
-	 * cannot call before it fit(Features *f)
-	 *
-	 */
-	bool init_randomcoefficients();
-
-	/** a getter
-	 * @return the set value of protected member dim_input_space
+	/** get output feature space dim
+	 * @return feature space dimension
 	 */
-	int32_t get_dim_input_space() const;
-
-	/** a getter
-	 * @return the set value of protected member dim_feature_space
-	 */
-	int32_t get_dim_feature_space() const;
+	int32_t get_dim_feature_space() const noexcept;
 
 	/** inherited from base class
 	 * does nothing
 	 */
 	void cleanup();
 
-	/// return the name of the preprocessor
 	virtual const char* get_name() const { return "RandomFourierGaussPreproc"; }
 
-	/// return a type of preprocessor
 	virtual EPreprocessorType get_type() const { return P_RANDOMFOURIERGAUSS; }
 
 protected:
 	/** default processing routine, inherited from base class
 	 * @param matrix the features matrix to be processed
-	 * @return the processed feature matrix from the DenseFeatures<float64_t>
-	 * class in case (2) (see description above) this routine requires only
-	 * steps 2a) and 2b), the rest is determined automatically
+	 * @return the processed feature matrix of class DenseFeatures<float64_t>
 	 */
 	virtual SGMatrix<float64_t> apply_to_matrix(SGMatrix<float64_t> matrix);
 
-	/**
-	 * helper for copy constructor and assignment operator=
-	 */
-	void copy(const RandomFourierGaussPreproc & feats); // helper for two constructors
-
-
-	/** dimension of input features
-	 * width of gaussian kernel in the form of exp(-x^2 / (2.0 kernelwidth^2) ) NOTE the 2.0 and the power ^2 !
-	 */
-	float64_t kernelwidth;
-
-	/** dimension of input features
-	 * width of gaussian kernel in the form of exp(-x^2 / (2.0 kernelwidth^2) ) NOTE the 2.0 and the power ^2 !
-	 */
-	float64_t cur_kernelwidth;
-
-	/** desired dimension of input features as set by void set_dim_input_space(const int32_t dim)
-	 *
+	/** Samples from the fourier transform of the kernel and stores into
+	 * the internal members, inherited from base class
```

Comment:
no need to state that something is inherited from base classes, instead use `override`

Reply:
but in this case it is not even inherited or am i wrong?

Reply:
Not yet ;), at least not until we make our superclass.

Reply:
so then why write this?

Reply:
No hard reason, for reference really. I'll move this docstring to the superclass when I make it, but I'll remove this now.

Reply:
as said before, docstrings should never contain things that are obvious from the code, which includes that a method is inherited or that is has to be overloaded by subclasses. There are keywords to make this clear and it is perfectly self explaining

Reply:
not resolved, and probably the ones below neither

Reply:
Fixed in local, kindly wait for the push. Please have some faith!